In [49]:
import pandas as pd

#Load the csv
df = pd.read_csv(r'C:\Users\tnh05\OneDrive\Desktop\Data Wrang\parsed.csv')

# Check the first few rows
df.head()

alert  cdi      code                                             detail  \
0   NaN  NaN  37389218  https://earthquake.usgs.gov/fdsnws/event/1/que...   
1   NaN  NaN  37389202  https://earthquake.usgs.gov/fdsnws/event/1/que...   
2   NaN  4.4  37389194  https://earthquake.usgs.gov/fdsnws/event/1/que...   
3   NaN  NaN  37389186  https://earthquake.usgs.gov/fdsnws/event/1/que...   
4   NaN  NaN  73096941  https://earthquake.usgs.gov/fdsnws/event/1/que...   

       dmin  felt    gap           ids   mag magType  ...     status  \
0  0.008693   NaN   85.0  ,ci37389218,  1.35      ml  ...  automatic   
1  0.020030   NaN   79.0  ,ci37389202,  1.29      ml  ...  automatic   
2  0.021370  28.0   21.0  ,ci37389194,  3.42      ml  ...  automatic   
3  0.026180   NaN   39.0  ,ci37389186,  0.44      ml  ...  automatic   
4  0.077990   NaN  192.0  ,nc73096941,  2.16      md  ...  automatic   

           time                          title tsunami        type  \
0  1.539480e+12  M 1.4 - 9km NE of Aguanga, CA       0  earthquake   
1  1.539480e+12  M 1.3 - 9km NE of Aguanga, CA       0  earthquake   
2  1.539480e+12  M 3.4 - 8km NE of Aguanga, CA       0  earthquake   
3  1.539470e+12  M 0.4 - 9km NE of Aguanga, CA       0  earthquake   
4  1.539470e+12  M 2.2 - 10km NW of Avenal, CA       0  earthquake   

                                               types     tz       updated  \
0         ,geoserve,nearby-cities,origin,phase-data, -480.0  1.539480e+12   
1         ,geoserve,nearby-cities,origin,phase-data, -480.0  1.539480e+12   
2  ,dyfi,focal-mechanism,geoserve,nearby-cities,o... -480.0  1.539540e+12   
3         ,geoserve,nearby-cities,origin,phase-data, -480.0  1.539480e+12   
4  ,geoserve,nearby-cities,origin,phase-data,scit... -480.0  1.539480e+12   

                                                 url parsed_place  
0  https://earthquake.usgs.gov/earthquakes/eventp...   California  
1  https://earthquake.usgs.gov/earthquakes/eventp...   California  
2  https://earthquake.usgs.gov/earthquakes/eventp...   California  
3  https://earthquake.usgs.gov/earthquakes/eventp...   California  
4  https://earthquake.usgs.gov/earthquakes/eventp...   California  

[5 rows x 27 columns]

# Problem 1

#### -selecting the appropriate columns and values while finding the 95th percentile using quantile()

In [52]:
df.columns

Index(['alert', 'cdi', 'code', 'detail', 'dmin', 'felt', 'gap', 'ids', 'mag',
       'magType', 'mmi', 'net', 'nst', 'place', 'rms', 'sig', 'sources',
       'status', 'time', 'title', 'tsunami', 'type', 'types', 'tz', 'updated',
       'url', 'parsed_place'],
      dtype='object')

In [54]:
df['mag'].unique()

array([ 1.35,  1.29,  3.42,  0.44,  2.16,  2.61,  1.7 ,  1.13,  0.92,
        4.7 ,  0.5 ,  2.77,  4.5 ,  2.13,  2.  ,  0.72,  1.69,  2.82,
        0.81,  0.55,  0.67,  2.1 ,  2.8 ,  1.1 ,  1.8 ,  0.93,  1.52,
        2.3 ,  1.3 ,  2.2 ,  1.9 ,  5.  ,  1.6 ,  2.12, -0.1 ,  1.31,
        0.2 ,  0.86,  2.92,  0.32,  1.84,  1.14,  0.9 ,  2.4 ,  2.5 ,
        1.46,  4.6 ,  2.21,  1.83,  2.15,  1.44,  2.07,  0.1 ,  2.76,
        2.24,  1.45,  0.35,  0.7 ,  2.43,  1.12,  1.5 ,  0.62,  2.01,
        1.4 ,  0.83,  0.  ,  3.22,  0.75,  1.57,  1.2 ,  0.59,  1.33,
        0.03,  1.64,  0.57,  2.81,  0.29,  2.84,  6.7 ,  2.36,  0.73,
        2.97,  2.09,  1.85,  0.4 ,  0.8 ,  2.7 , -0.4 ,  0.33,  2.49,
        3.  ,  0.89,  4.2 ,  0.6 ,  2.9 ,  0.58,  0.74,  1.38,  0.61,
        0.65, -0.02,  0.88,  1.41,  1.08,  0.68,  5.2 ,  1.75,  0.54,
        1.72,  1.36,  1.  ,  1.16,  0.56,  0.82, -0.2 ,  1.01,  0.87,
        1.87,  2.33,  3.29,  0.52,  1.03,  2.83,  3.07,  4.4 ,  5.7 ,
        4.9 ,  1.92,

In [56]:
# Filtering earthquakes with mg magnitude types
japan_earthquakes_mb = df[(df['parsed_place'] == 'Japan') & (df['magType'] == 'mb')]

# Calculating the 95th percentile of magnitude
percentile_95 = japan_earthquakes_mb['mag'].quantile(0.95)
print(percentile_95)

4.9


### 4.9 is the 95th percentile of mb type of earthquakes in Japan

# Problem 2

#### Finding the percentage of earthquakes in Indonesia that were coupled with tsunamis

In [59]:
# Filtering earthquakes within Indonesia
indonesia_earthquakes = df[df['parsed_place'] == 'Indonesia']

# Count the total number of earthquakes coupled with tsunamis
total_earthquakes_indonesia = len(indonesia_earthquakes)
tsunami_earthquakes = len(indonesia_earthquakes[indonesia_earthquakes['tsunami'] == 1])

# Calculate the percentage of earthquakes coupled with tsunamis
percentage_tsunami = (tsunami_earthquakes / total_earthquakes_indonesia) * 100
print(f"Percentage of earthquakes in Indonesia coupled with tsunamis: {percentage_tsunami:.2f}%")

Percentage of earthquakes in Indonesia coupled with tsunamis: 23.13%


In [60]:
# Filtering earthquakes within Indonesia
indonesia_earthquakes = df[df['parsed_place'] == 'Indonesia']

# Count the total number of earthquakes coupled with tsunamis
total_earthquakes_indonesia = len(indonesia_earthquakes)
tsunami_earthquakes = len(indonesia_earthquakes[indonesia_earthquakes['tsunami'] == 1])

# Calculate the percentage of earthquakes coupled with tsunamis
percentage_tsunami = (tsunami_earthquakes / total_earthquakes_indonesia) * 100
print(f"Percentage of earthquakes in Indonesia coupled with tsunamis: {percentage_tsunami:.2f}%")

Percentage of earthquakes in Indonesia coupled with tsunamis: 23.13%


# Problem 3

#### Calculate summary statistics for earthquakes in Nevada

In [70]:
# Using describe() to check out the summary of the Nevada value in the parsed_place columns
df[df.parsed_place == 'Nevada'].describe()

cdi        dmin       felt         gap         mag   mmi  \
count  15.000000  681.000000  15.000000  681.000000  681.000000  1.00   
mean    2.440000    0.166199   2.400000  153.668120    0.500073  2.84   
std     0.501142    0.166228   4.626013   68.735302    0.696710   NaN   
min     2.000000    0.001000   1.000000   29.140000   -0.500000  2.84   
25%     2.000000    0.053000   1.000000   97.380000   -0.100000  2.84   
50%     2.200000    0.112000   1.000000  149.140000    0.400000  2.84   
75%     2.900000    0.233000   1.000000  199.720000    0.900000  2.84   
max     3.300000    1.414000  19.000000  355.910000    2.900000  2.84   

              nst         rms         sig          time  tsunami     tz  \
count  681.000000  681.000000  681.000000  6.810000e+02    681.0  681.0   
mean    12.618209    0.151986   10.970631  1.538314e+12      0.0 -480.0   
std      9.866963    0.084662   19.607150  5.966149e+08      0.0    0.0   
min      3.000000    0.000500    0.000000  1.537250e+12      0.0 -480.0   
25%      6.000000    0.106900    0.000000  1.537850e+12      0.0 -480.0   
50%     10.000000    0.146300    2.000000  1.538280e+12      0.0 -480.0   
75%     16.000000    0.187100   12.000000  1.538820e+12      0.0 -480.0   
max     61.000000    0.863400  129.000000  1.539460e+12      0.0 -480.0   

            updated  
count  6.810000e+02  
mean   1.538401e+12  
std    6.012149e+08  
min    1.537310e+12  
25%    1.537930e+12  
50%    1.538430e+12  
75%    1.538880e+12  
max    1.539480e+12

# Problem 4

#### Calculate summary statistics for earthquakes in Nevada

In [81]:
# code creates a new coloumn ring of fire location that checks in the appropriate column accesing
# the values with str and using the function contains() then joining a set of vaulues
df['ring_of_fire_location'] = df.parsed_place.str.contains(r'|'.join(['Alaska', 'Antarctic', 'Bolivia', 'California', 'Canada','Chile', 'Costa Rica', 'Ecuador', 'Fiji', 'Guatemala',
'Indonesia', 'Japan', 'Kermadec Islands', '^Mexico','New Zealand', 'Peru', 'Philippines', 'Russia','Taiwan', 'Tonga', 'Washington']))

# Problem 5

#### Calculate the number of earthquakes in the ring of fire locations and the number outside

In [88]:
# using value_counts() wihtout any paramaters
df.ring_of_fire_location.value_counts()

ring_of_fire_location
True     7188
False    2144
Name: count, dtype: int64

# Problem 6

#### Find the tsunami count along the ring of fire

In [93]:
# -using sum() to find the count of tsunamis
df.loc[df.ring_of_fire_location, 'tsunami'].sum()

45

### Title: 3.2 Exercise
### Author: Stefanie Molin
### Date: 09/13/2024
### Modified By: Vanessa Williams
### Description: Using to add a column containing specific locations